### ISSUE TITLE
Azure Queue Storage returns messages with invalid dates 

### ISSUE THREAD

**Issue by sarintom** [arafato/azurite#172]
I'm using Dockerized Azurite to test local Java application. I'm connecting to the Azurite with official Azure Storage Java SDK (https://github.com/Azure/azure-storage-java). When adding messages to the local test queue the SDK hangs in a retry loop since it cannot parse the newly added messages timestamps. Same thing happens when reading the queue with the SDK.

I traced the error to be with the time notation containing garbage in front of the date time string. When requesting the message via HTTP call there seems to be nothing wrong with the dates:

```
 <InsertionTime>Wed, 21 Mar 2018 08:17:40 GMT</InsertionTime>
```

But when the Azure Storage Java SDK receives that date it has a line break and some spaces in front of the date:

```
"\n        Wed, 21 Mar 2018 08:17:40 GMT"
```

When the Java SDK reads from the real cloud Azure Storage Queue the dates are correct and no exception is thrown when parsing:

```
"Wed, 21 Mar 2018 08:06:22 GMT"
```

Java Exception thrown when parsing the invalid date is (QueueMessageHandler.java:109):

```
java.text.ParseException: Unparseable date: "
        Wed, 21 Mar 2018 08:17:40 GMT"
```

I submit this issue here instead of the Azure Storage Java SDK since it seems to be working correctly with the actual cloud Azure Storage Queue.

**Comment by sarintom** [comment]
Actually it seems that the official Azure SDK uses string builder to read the xml data in. It reads data from the XML response character at a time and somehow it adds the line change and spaces from the XML response pretty printing. So the insertionTime includes the line change `\n` after `</MessageId>` element and the indention of 8 spaces before the `<InsertionTime>` element. So the insertion time becomes

```
"\n        Tue, 20 Mar 2018 14:24:49 GMT"
```

Here's the example XML response about a message:

```
<?xml version='1.0'?>
<QueueMessagesList>
    <QueueMessage>
        <MessageId>72c0b0c6-508f-400a-adfc-19d72820ba2c</MessageId>
        <InsertionTime>Tue, 20 Mar 2018 14:24:49 GMT</InsertionTime>
        <ExpirationTime>Tue, 27 Mar 2018 14:24:49 GMT</ExpirationTime>
        <PopReceipt>3ee0d952-be0d-42d2-b9c8-54f6f829b3ad</PopReceipt>
        <TimeNextVisible>Wed, 21 Mar 2018 08:39:13 GMT</TimeNextVisible>
        <DequeueCount>1</DequeueCount>
        <MessageText>bmFhbWFiYWFtYQ==</MessageText>
    </QueueMessage>
</QueueMessagesList>
````

So maybe this can be fixed by returning the XML data without pretty printing?

**Comment by sarintom** [comment]
Created a pull request for removing the pretty printing of XML formatting:

https://github.com/arafato/azurite/pull/173

Tried it with the Java SDK and seems to fix the issue.

**Comment by arafato** [comment]
@sarintom thanks so much for reporting this! I've also just committed a fix which simply uses a regex to remove whitespace characters and linebreaks. Released as `v2.6.4`. Can you check if it works for you? If not I will apply your PR. I wasn't aware of the pretty printing option...

**Comment by sarintom** [comment]
Thanks, seems to be working this way also. Is the fix automatically available through a new Docker image from Dockerhub?

**Comment by arafato** [comment]
Yes, the new version is already available and tagged as `2.6.4` and `latest`
________________________________
